exportReport/Export/ExportReport.java

```diff
@@ -0,0 +1,255 @@
+import java.io.File;
+import java.io.UnsupportedEncodingException;
+import java.text.MessageFormat;
+import java.util.Locale;
+
+import javax.swing.JOptionPane;
+import javax.swing.SwingWorker;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.ViewDelegate;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.extension.exportReport.ExtensionExportReport;
+import org.zaproxy.zap.extension.exportReport.FileChooser.*;
+import org.zaproxy.zap.utils.DesktopUtils;
+import org.zaproxy.zap.view.widgets.WritableFileChooser;
+
+/*
+ * AUTHOR		:	GORAN SARENKAPA - JordanGS
+ * SPONSOR		:	RYERSON UNIVERSITY
+ * CLASS		:	ExportReport.java 
+ * DESC			:	FileChooser setup class. Runs the Export as a background task with a Swing Worker.
+ * CREATED ON	:	MARCH 10TH, 2016
+ * CURRENT VER	:	V1.0
+ * SOURCE		:	https://github.com/JordanGS/workspace/tree/master/zap-extensions/src/org/zaproxy/zap/extension/exportReport
+ */
+
+/* 
+ * MODIFED BY	:	<NAME> - <GIT USER>
+ * MOD DATE		:	
+ * MOD VERSION	:	<VERSION OF PLUGIN>
+ * MOD DESC		:	
+ */
+
+public class ExportReport {
+	private Logger logger = Logger.getLogger(ExportReport.class);
+
+	private Task task; // Create a task and dump the export into the background,
+						// essential for large exports.
+	private WritableFileChooser fc; // Global because the property change
+									// listener won't be able to access if it's
+									// local scope.
+
+	class Task extends SwingWorker<Void, Void> {
+		private ViewDelegate view;
+		private ExtensionExportReport extension;
+		private String fileExtension;
+
+		public Task(ViewDelegate view, ExtensionExportReport extension, String fileExtension) {
+			this.view = view;
+			this.extension = extension;
+			this.fileExtension = fileExtension;
+		}
+
+		/**
+		 * Main task. Executed in background thread.
+		 */
+		@Override
+		public Void doInBackground() {
+			File f = fc.getSelectedFile();
+
+			String file = f.getName();
+			String absolutePath = f.getAbsolutePath();
+			String fileName = file.substring(0, file.lastIndexOf(fileExtension) - 1);
+			String path = absolutePath.substring(0, absolutePath.lastIndexOf(file));
+
+			String XML_PATH = "";
+			try
+			{
+				XML_PATH = ReportExport.generateDUMP(path, fileName, extension.extensionGetTitle(),
+						extension.extensionGetBy(), extension.extensionGetFor(), extension.extensionGetScanDate(),
+						extension.extensionGetScanVer(), extension.extensionGetReportDate(),
+						extension.extensionGetReportVer(), extension.getDescription(),
+						extension.getIncludedAlertSeverity(), extension.getIncludedAlertDetails());
+			}
+			catch (UnsupportedEncodingException e)
+			{
+				logger.error(e.getMessage(), e);
+				view.showMessageDialog(MessageFormat.format("Error: Problem encoding the DUMP File content, please see log for more details.", new Object[] { absolutePath }));
```

Comment:
Unintended misuse of the `MessageFormat.format`, mistake carried over from previous version. The fix is to remove it.